src/org/zaproxy/zap/extension/proxies/OptionsLocalProxyPanel.java

```diff
@@ -0,0 +1,316 @@
+            gridBagConstraints2.gridy = 2;
+            gridBagConstraints2.anchor = java.awt.GridBagConstraints.NORTHWEST;
+            gridBagConstraints2.fill = java.awt.GridBagConstraints.HORIZONTAL;
+            gridBagConstraints2.insets = new java.awt.Insets(2, 2, 2, 2);
+            gridBagConstraints2.weightx = 1.0D;
+
+            panelProxy.add(getPanelLocalProxy(), gridBagConstraints81);
+            gridBagConstraints81.gridy = 1;
+        }
+        return panelProxy;
+    }
+
+    private JComboBox<String> getLocalAddress() {
+        if (localAddress == null) {
+            localAddress = new JComboBox<String>();
+            List<String> allAddrs = NetworkUtils.getAvailableAddresses(false);
+            localAddress.addItem("localhost");
+            localAddress.addItem("127.0.0.1");
+            localAddress.addItem("::1");
+            localAddress.addItem("0.0.0.0");
+            for (String addr : allAddrs) {
+                localAddress.addItem(addr);
+            }
+        }
+        return localAddress;
+    }
+
+    private JCheckBox getChkBehindNat() {
+        if (chkBehindNat == null) {
+            chkBehindNat = new JCheckBox(Constant.messages.getString("options.proxy.local.label.behindnat"));
+            chkBehindNat.setToolTipText(Constant.messages.getString("options.proxy.local.tooltip.behindnat"));
+        }
+        return chkBehindNat;
+    }
+
+    public JCheckBox getChkRemoveUnsupportedEncodings() {
+        if (chkRemoveUnsupportedEncodings == null) {
+            chkRemoveUnsupportedEncodings = new JCheckBox(
+                    Constant.messages.getString("options.proxy.local.label.removeUnsupportedEncodings"));
+            chkRemoveUnsupportedEncodings
+                    .setToolTipText(Constant.messages.getString("options.proxy.local.tooltip.removeUnsupportedEncodings"));
+        }
+        return chkRemoveUnsupportedEncodings;
+    }
+
+    private JCheckBox getChkAlwaysDecodeGzip() {
+        if (chkAlwaysDecodeGzip == null) {
+            chkAlwaysDecodeGzip = new JCheckBox(Constant.messages.getString("options.proxy.local.label.alwaysDecodeGzip"));
+            chkAlwaysDecodeGzip.setToolTipText(Constant.messages.getString("options.proxy.local.tooltip.alwaysDecodeGzip"));
+        }
+        return chkAlwaysDecodeGzip;
+    }
+
+    /**
+     * This method initializes spinnerProxyPort
+     *
+     * @return ZapPortNumberSpinner
+     */
+    private ZapPortNumberSpinner getSpinnerProxyPort() {
+        if (spinnerProxyPort == null) {
+            // Do not allow invalid port numbers
+            spinnerProxyPort = new ZapPortNumberSpinner(8080);
+        }
+        return spinnerProxyPort;
+    }
+
+    private void initialize() {
+        this.setLayout(new CardLayout());
+        this.setName(Constant.messages.getString("options.proxy.local.title"));
+        if (Model.getSingleton().getOptionsParam().getViewParam().getWmUiHandlingOption() == 0) {

```

Comment:
Can be removed? (this panel would be added to other panels which are already properly sized?)